sim/hunter/kill_shot.go

```diff
@@ -0,0 +1,76 @@
+package hunter
+
+import (
+	"time"
+
+	"github.com/wowsims/sod/sim/core"
+	"github.com/wowsims/sod/sim/core/proto"
+)
+
+func (hunter *Hunter) registerKillShotSpell() {
+	if !hunter.HasRune(proto.HunterRune_RuneLegsKillShot) {
+		return
+	}
+
+	baseDamage := 113 / 100 * 100 / 100 * (2.976264 + 0.641066 * float64(hunter.Level) + 0.022519 * float64(hunter.Level) * float64(hunter.Level)) // taken from wowhead
+
+	hasCobraStrikes := hunter.pet != nil && hunter.HasRune(proto.HunterRune_RuneChestCobraStrikes)
+
+	// Efficiency talent doesn't apply to this spell even though it has 'shot' in the name
+	manaCostMultiplier := 1.0
+	if hunter.HasRune(proto.HunterRune_RuneChestMasterMarksman) {
+		manaCostMultiplier -= 0.25
+	}
+
+	hunter.KillShot = hunter.RegisterSpell(core.SpellConfig{
+		ActionID:     core.ActionID{SpellID: 409593},
+		SpellSchool:  core.SpellSchoolPhysical,
+		DefenseType:  core.DefenseTypeRanged,
+		ProcMask:     core.ProcMaskRangedSpecial,
+		Flags:        core.SpellFlagMeleeMetrics | core.SpellFlagAPL,
+		CastType:     proto.CastType_CastTypeRanged,
+		MissileSpeed: 24,
+
+		ManaCost: core.ManaCostOptions{
+			BaseCost:   0.03,
+			Multiplier: manaCostMultiplier,
+		},
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				GCD: core.GCDDefault,
+			},
+			IgnoreHaste: true, // Hunter GCD is locked at 1.5s
```

Comment:
This shouldn't actually matter. We removed DoT and GCD haste contribution from Wrath. TBH these options could probably be removed but I figure keeping them around is safer just in case

Reply:
Do you want me to remove them or just leave them in?